internal/service/cloudbackupsnapshotexportjob/resource_cloud_backup_snapshot_export_job_test.go

```diff
@@ -120,45 +119,107 @@ func readRequired(rs *terraform.ResourceState, resourceName string) (projectID,
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": "s3:GetBucketLocation",
+            "Resource": "arn:aws:s3:::%[2]s"
+        },
+        {
+            "Effect": "Allow",
+            "Action": "s3:PutObject",
+            "Resource": "arn:aws:s3:::%[2]s/*"
+        }]
+    }
+    EOF
+}
+
+resource "aws_iam_role" "test_role" {
+    name = %[3]q
+    assume_role_policy = <<EOF
+	{
+	  "Version": "2012-10-17",
+	  "Statement": [
+	    {
+	      "Effect": "Allow",
+	      "Principal": {
+	        "AWS": "${mongodbatlas_cloud_provider_access_setup.setup_only.aws_config[0].atlas_aws_account_arn}"
+	      },
+	      "Action": "sts:AssumeRole",
+	      "Condition": {
+	        "StringEquals": {
+	          "sts:ExternalId": "${mongodbatlas_cloud_provider_access_setup.setup_only.aws_config[0].atlas_assumed_role_external_id}"
+	        }
+	      }
+	    }
+	  ]
+	}
+	EOF
+}
+
+resource "aws_s3_bucket" "backup" {
+	bucket          = %[2]q
+	force_destroy   = true
+}
+
+resource "mongodbatlas_cloud_provider_access_setup" "setup_only" {
+	project_id    = %[1]q
+	provider_name = "AWS"
+}
+
+resource "mongodbatlas_cloud_provider_access_authorization" "auth_role" {
+	project_id = %[1]q
+	role_id    = mongodbatlas_cloud_provider_access_setup.setup_only.role_id
+	aws {
+	  iam_assumed_role_arn = aws_iam_role.test_role.arn
+	}
+}
+
 resource "mongodbatlas_cloud_backup_snapshot" "test" {
-  project_id        = %[1]q
-  cluster_name      = %[4]q
-  description       = "tf-acc-test"
-  retention_in_days = 1
+	project_id        = %[1]q
+	cluster_name      = %[5]s
+	description       = "tf-acc-test"
+	retention_in_days = 1
 }
 
 resource "mongodbatlas_cloud_backup_snapshot_export_bucket" "test" {
-  project_id     = %[1]q
-  iam_role_id    = "%[3]s"
-  bucket_name    = "%[2]s"
-  cloud_provider = "AWS"
+	project_id     = %[1]q
+	iam_role_id    = mongodbatlas_cloud_provider_access_authorization.auth_role.role_id
+	bucket_name    = aws_s3_bucket.backup.bucket
+	cloud_provider = "AWS"
 }
 
 resource "mongodbatlas_cloud_backup_snapshot_export_job" "test" {
-  project_id   		= %[1]q
-  cluster_name 		= %[4]q
-  snapshot_id		= mongodbatlas_cloud_backup_snapshot.test.snapshot_id
-  export_bucket_id 	= mongodbatlas_cloud_backup_snapshot_export_bucket.test.export_bucket_id
-
-  custom_data {
-    key   = "exported by"
-    value = "tf-acc-test"
-  }
+	project_id   		= %[1]q
+	cluster_name 		= %[5]s
+	snapshot_id		= mongodbatlas_cloud_backup_snapshot.test.snapshot_id
+	export_bucket_id 	= mongodbatlas_cloud_backup_snapshot_export_bucket.test.export_bucket_id
+	custom_data {
+		key   = "exported by"
+		value = "tf-acc-test"
+	}
 }
 
 data "mongodbatlas_cloud_backup_snapshot_export_job" "test" {
     project_id 		= %[1]q
-    cluster_name 	= %[4]q
+    cluster_name 	= %[5]s
     export_job_id 	= mongodbatlas_cloud_backup_snapshot_export_job.test.export_job_id
+    id              = "any_value"
```

Comment:
I guess this can be removed after the next release?

Reply:
BTW: I think the implementation was wrong before afbd7b7dd203d1e4ea5fa5ad90b767ea080e59e5
Therefore, the migration test will probably never pass until the next release...

Reply:
yes, we can remove it. let me leave a comment.
The migration test (at some point, if we don't count the timeouts) passed once I made this change. Before they were not being executed so it is possible it was failing and never realised.

Let me know if I am missing something

Reply:
Ok. I would be quite surprised if the `TestMigBackupSnapshotExportJob_basic` passes. 
Please share a link if you find the action where it succeeded 👍

Reply:
You were right about this - this is what I believe happened:
1. I tested it locally and I am pretty sure it passed (I am not sure why)
2. During PR tests, a lot of networking issues were happening randomly on different tests. I believe that must have hidden this problem
3. This morning the networking issues went away and the test started failing for its right reason. 

Going to publish another PR to disable the migration test until 1.16.1